internal/provider/resources/flow.go

```diff
@@ -0,0 +1,372 @@
+
+	var tags []string
+	resp.Diagnostics.Append(model.Tags.ElementsAs(ctx, &tags, false)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.Flows(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating flows client",
+			fmt.Sprintf("Could not create flows client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
+	}
+
+	flow, err := client.Create(ctx, api.FlowCreate{
+		Name: model.Name.ValueString(),
+		Tags: tags,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating flow",
+			fmt.Sprintf("Could not create flow, unexpected error: %s", err),
+		)
+
+		return
+	}
+
+	resp.Diagnostics.Append(copyFlowToModel(ctx, flow, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *FlowResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var model FlowResourceModel
+
+	// Populate the model from state and emit diagnostics on error
+	resp.Diagnostics.Append(req.State.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// if model.ID.IsNull() && model.Handle.IsNull() {
+	// 	resp.Diagnostics.AddError(
+	// 		"Both ID and Handle are unset",
+	// 		"This is a bug in the Terraform provider. Please report it to the maintainers.",
+	// 	)
+
+	// 	return
+	// }
+
+	client, err := r.client.Flows(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating deployment client",
+			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
+	}
+
+	// A deployment can be imported + read by either ID or Handle
+	// If both are set, we prefer the ID
+	var flow *api.Flow
+	if !model.ID.IsNull() {
+		var flowID uuid.UUID
+		flowID, err = uuid.Parse(model.ID.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddAttributeError(
+				path.Root("id"),
+				"Error parsing Flow ID",
+				fmt.Sprintf("Could not parse Flow ID to UUID, unexpected error: %s", err.Error()),
+			)
+
+			return
+		}
+
+		flow, err = client.Get(ctx, flowID)
+	}
+	// } else if !model.Handle.IsNull() {
+	// 	var deployments []*api.Deployment
+	// 	deployments, err = client.List(ctx, []string{model.Handle.ValueString()})
+
+	// 	// The error from the API call should take precedence
+	// 	// followed by this custom error if a specific deployment is not returned
+	// 	if err == nil && len(deployments) != 1 {
+	// 		err = fmt.Errorf("a deployment with the handle=%s could not be found", model.Handle.ValueString())
+	// 	}
+
+	// 	if len(deployments) == 1 {
+	// 		deployment = deployments[0]
+	// 	}
+	// }
```

Comment:
What's the plan for this part of the logic?

Reply:
no plan, there is no need for this code. removed.